atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java

```diff
@@ -220,6 +223,16 @@ public void testStoreByteStreamFiveMegaBytes_compressedStream_incompressible() t
         storeAndCheckByteStreams(compressedStore, getIncompressibleBytes(5_000_000));
     }
 
+    @Test
+    public void testStoreToMaxMemStream() throws IOException {
+        storeAndCheckByteStreams(maxMemStore, getIncompressibleBytes(100));
+    }
+
+    @Test
+    public void testStoreHugeToMaxMemStream() throws IOException {
+        storeAndCheckByteStreams(maxMemStore, getIncompressibleBytes(20_000_000));
```

Comment:
we should add fence posts tests for this and other stores for at least `+1` & `-1` around block boundaries.